.chloggen/fix_k8s_metrics.yaml

```diff
@@ -0,0 +1,22 @@
+# Use this changelog template to create an entry for release notes.
+#
+# If your change doesn't affect end users you should instead start
+# your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the area of concern in the attributes-registry, (e.g. http, cloud, db)
+component: k8s
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Change k8s.replication_controller metrics to k8s.replicationcontroller
```

Comment:
sorry I missed the discussion in https://github.com/open-telemetry/semantic-conventions/pull/1742#discussion_r1926306420

I feel like this opens a pretty big door across semconv for debating when to apply underscores, because we could point to lots of domain-specific apis that concat words instead of underscoring them

I understand (and agree) with `replicaset`, `statefulset`, `daemonset`, even outside of the k8s domain. could we just make an exception for those (`*set` things)?

Reply:
to me this boils down to https://github.com/open-telemetry/semantic-conventions/issues/1755#issuecomment-2610956092: k8s uses `replicationcontroller` for everything and we're being consistent with the domain.

Similarly to https://github.com/open-telemetry/semantic-conventions/blob/b865f63bc7ba3039ad87504d75ee40104149d1bd/model/azure/registry.yaml#L78-L80

there is some external source of this value and we pick this external spelling for attributes limited to this use-case.

Reply:
the benefit of doing this for attribute _values_ seems clear to me (e.g. avoids instrumentations having to map domain-specific values to something else)

the benefit to doing this for attribute _names_ is less clear to me

Reply:
how would we approach something like `replicaset`, `statefulset`, `daemonset`  vs `replication_controller` in the future? There are some other examples in https://github.com/open-telemetry/semantic-conventions/pull/1742#discussion_r1926519917 like `cronjob` or `apiservice` where we'd have ambiguity

Reply:
also `mariadb` and `cosmosdb` could arguably be `maria_db` and `cosmos_db`. I'm not suggesting it, I'm saying that I genuinely don't know when and why we apply `_`